### PR TITLE
restructured error handling in georocket configuration

### PR DESCRIPTION
**Pull request by andrej-sajenko**

### CODE REVIEW

File: georocket-server/src/main/java/io/georocket/GeoRocket.java [hunk 1]
```diff
@@ -552,52 +552,54 @@ protected static String getGeoRocketHomeStr() {
    * Get the GeoRocket home as {@link File}
    * @param geoRocketHomeStr the path to the GeoRocket home
    * @return the GeoRocket home file
+   * @throws IOException if
```
Comment (michel-kraemer) [inline georocket-server/src/main/java/io/georocket/GeoRocket.java]: if ... ?

File: georocket-server/src/main/java/io/georocket/GeoRocket.java [hunk 2]
```diff
@@ -552,52 +552,54 @@ protected static String getGeoRocketHomeStr() {
...
+    return new File(geoRocketHomeStr).getCanonicalFile();
   }
 
   /**
    * Load the GeoRocket configuration and return it.
    * Will return an empty configuration if something went wrong.
    * @param geoRocketHome the GeoRocket home path
    * @return the configuration
+   *
+   * @throws IllegalStateException If the configuration file could not be found
```
Comment (michel-kraemer) [inline georocket-server/src/main/java/io/georocket/GeoRocket.java]: Does it really throw an IllegalStateException?

File: georocket-server/src/main/java/io/georocket/GeoRocket.java [hunk 3]
```diff
@@ -552,52 +552,54 @@ protected static String getGeoRocketHomeStr() {
...
+   * @throws DecodeException If failed to decode the JSON configuration
```
Comment (michel-kraemer) [inline georocket-server/src/main/java/io/georocket/GeoRocket.java]: I suppose DecodeException is an IOException? Please check.

File: georocket-server/src/main/java/io/georocket/GeoRocket.java [hunk 4]
```diff
@@ -552,52 +552,54 @@ protected static String getGeoRocketHomeStr() {
...
    */
-  protected static JsonObject loadConfiguration(File geoRocketHome) {
+  protected static JsonObject loadConfiguration(File geoRocketHome) throws IOException, DecodeException {
     File confDir = new File(geoRocketHome, "conf");
     File confFile = new File(confDir, "georocketd.json");
 
     try {
       String confFileStr = FileUtils.readFileToString(confFile, "UTF-8");
       return new JsonObject(confFileStr);
     } catch (IOException e) {
```
Comment (michel-kraemer) [inline georocket-server/src/main/java/io/georocket/GeoRocket.java]: Catching an IOException and then wrapping it in another one doesn't make much sense.

File: georocket-server/src/main/java/io/georocket/GeoRocket.java [hunk 5]
```diff
@@ -552,52 +552,54 @@ protected static String getGeoRocketHomeStr() {
...
-    
-    return new JsonObject();
   }
 
   /**
    * Get and load the GeoRocket configuration
    * @return the configuration
    */
-  protected static JsonObject configureGeoRocket() {
+  protected static JsonObject configureGeoRocket() throws Exception {
```
Comment (michel-kraemer) [inline georocket-server/src/main/java/io/georocket/GeoRocket.java]: Please use the exception that is really thrown. Add javadoc.

File: georocket-server/src/main/java/io/georocket/GeoRocket.java [hunk 6]
```diff
@@ -613,18 +615,22 @@ protected static JsonObject configureGeoRocket() {
    * @param args the command line arguments
    */
   public static void main(String[] args) {
-    JsonObject conf = configureGeoRocket();
+    try {
```
Comment (michel-kraemer) [inline georocket-server/src/main/java/io/georocket/GeoRocket.java]: You only need to put the `configureGeoRocket` call in the try-catch block. The rest can be outside the block.

**Comment by michel-kraemer** [comment]
Many comments for such a small commit. Please be more careful. Will save us all a lot of time.

File: georocket-server/src/main/java/io/georocket/GeoRocket.java (hunk 3 above)
Comment (andrej-sajenko) [inline georocket-server/src/main/java/io/georocket/GeoRocket.java]: DecodeException is a RuntimeException

**Pull request closed without merging**